### PR TITLE
Allow overriding branch name with ROCM_DOCS_REMOTE_DETAILS

### PR DESCRIPTION
**Pull request by Maetveis**
There are cases when the automatic branch name detection fails: i.e. when running on a commit not part of any branch. This could be e.g. a merge request pipeline that runs on a temporary merge commit.

For these situations allow overriding the automatic detection with an environment variable
`ROCM_DOCS_REMOTE_DETAILS=<remote-url>,<branch-name>,<allow-edits>`.

Additionally support local development on a branch without a remote, in this case disable editing, but return the current branch name.

### CODE REVIEW

**Comment by Maetveis** [comment]
@saadrahim I cant seem to be able to add reviewers, can you check this out? This is blocking our internal CI for hipCUB.

File: pyproject.toml
Comment (saadrahim) [inline pyproject.toml]: 0.1.0, as per semver

**Review by lawruble13 — APPROVED** [review]

**Review by saadrahim — APPROVED** [review]

**Comment by Maetveis** [comment]
Regarding the conflict: Should I bump the bugfix number as was done in #25 or the minor as it corresponds to semver, because this is a new feature (but we're also before 1.0, so no real rules apply).

**Comment by saadrahim** [comment]
> Regarding the conflict: Should I bump the bugfix number as was done in #25 or the minor as it corresponds to semver, because this is a new feature (but we're also before 1.0, so no real rules apply).

follow semver convention at all times please. I'd say this is as new feature, it should be at 0.1.0.

**Comment by saadrahim** [comment]
@Maetveis please merge after fixing merge conflict, use with 0.1.0.

**Comment by Maetveis** [comment]
> @Maetveis please merge after fixing merge conflict, use with 0.1.0.

Resolved the conflict, but I don't seem to have merge access.

**Pull request merged**
